server/src/domain/library/library.service.ts

```diff
@@ -384,17 +377,12 @@ export class LibraryService {
 
     this.logger.debug(`Found ${crawledAssetPaths.length} assets when crawling import paths ${library.importPaths}`);
     const assetsInLibrary = await this.assetRepository.getByLibraryId([job.id]);
-    const offlineAssets = assetsInLibrary.filter((asset) => !crawledAssetPaths.includes(asset.originalPath));
-    this.logger.debug(`${offlineAssets.length} assets in library are not present on disk and will be marked offline`);
+    const offlineAssetIds = assetsInLibrary
```

Comment:
Everything looks great, except this line which can be made into O(n) instead of O(n^2).